File: 1802-number-of-students-unable-to-eat-lunch/number-of-students-unable-to-eat-lunch.py
class Solution(object):
    def countStudents(self, students, sandwiches):
        """
        :type students: List[int]
        :type sandwiches: List[int]
        :rtype: int
        """
        c1= 0
        c0=0 
        for i in students:
            if(i):
                c1+=1
            else:
                c0+=1
        for i in sandwiches:
            if(i):
                if c1==0: 
                    break
                c1-=1
            else:
                if c0==0: 
                    break
                c0-=1
            
        return c1+c0

File: 1802-number-of-students-unable-to-eat-lunch/number-of-students-unable-to-eat-lunch.cpp
class Solution {
public:
    int countStudents(vector<int>& students, vector<int>& sandwiches) {
        int c1= 0, c0=0;
        for(int i : students){
            if(i)
                c1++;
            else
                c0++;
        }
        for(int i : sandwiches){
            if(i){
                if(c1==0) break;
                c1--;
            }else{
                if(c0==0) break;
                c0--;
            }
        }
        return c1+c0;
    }
};